components/builder-originsrv/src/data_store.rs

```diff
@@ -907,9 +939,37 @@ impl DataStore {
         package.set_exposes(exposes);
         package.set_deps(self.into_idents(row.get("deps")));
         package.set_tdeps(self.into_idents(row.get("tdeps")));
+
+        let channels: Vec<String> = row.get("channels");
+        let mut channels_buf = ::protobuf::RepeatedField::new();
+
+        for channel in channels {
+            channels_buf.push(channel);
+        }
+
+        package.set_channels(channels_buf);
         package
     }
 
+    fn row_to_package_ident_for_version(
+        &self,
```

Comment:
Yeah, I figured since it's part of an implementation, `&self` had to be there, but I guess unless we could also just change the way it gets called. I don't have a strong opinion of course, although it does look like this pattern is pretty common in our codebase, so I'd tend to leave it in there just for consistency, at least until we do have a more generic approach to these.